front/src/applications/stdcmV2/hooks/useStdmcEnv.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfActions } from 'common/osrdContext';
+
+export const NO_CONFIG_FOUND_MSG = 'No configuration found';
+
+export default function useStdcmEnvironment() {
+  const dispatch = useDispatch();
+  const {
+    updateInfraID,
+    updateTimetableID,
+    updateElectricalProfileSetID,
+    updateWorkScheduleGroupID,
+    updateSearchDatetimeWindow,
+  } = useOsrdConfActions();
+  const [getStdcmSearchEnvironment] =
+    osrdEditoastApi.endpoints.getStdcmSearchEnvironment.useLazyQuery();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<null | Error>(null);
+
+  const loadStdcmEnvironment = useCallback(async () => {
+    try {
+      setError(null);
+      setLoading(true);
+      const { data } = await getStdcmSearchEnvironment();
+      if (!data) throw new Error(NO_CONFIG_FOUND_MSG);
+      dispatch(updateInfraID(data.infra_id));
+      dispatch(updateTimetableID(data.timetable_id));
+      dispatch(updateElectricalProfileSetID(data.electrical_profile_set_id));
+      dispatch(updateWorkScheduleGroupID(data.work_schedule_group_id));
+      dispatch(
+        updateSearchDatetimeWindow([
+          new Date(data.search_window_begin),
+          new Date(data.search_window_end),
+        ])
+      );
```

Comment:
Can you create a unique action `loadStdcmEnvironment` (you can change the name) to update all the store at the same time ?